google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkCommitBatch.java

```diff
@@ -29,21 +26,13 @@
     super(firestore, maxBatchSize);
   }
 
-  BulkCommitBatch(
-      FirestoreImpl firestore,
-      BulkCommitBatch retryBatch,
-      final Set<DocumentReference> docsToRetry) {
+  BulkCommitBatch(FirestoreImpl firestore, BulkCommitBatch retryBatch) {
     super(firestore);
-    this.writes.addAll(
-        FluentIterable.from(retryBatch.writes)
-            .filter(
-                new Predicate<WriteOperation>() {
-                  @Override
-                  public boolean apply(WriteOperation writeOperation) {
-                    return docsToRetry.contains(writeOperation.documentReference);
-                  }
-                })
-            .toList());
+
+    // Create a new BulkCommitBatch containing only the indexes from the provided indexes to retry.
```

Comment:
Hm. This is different from Node, but likely better. Should we update the Node SDK to match? This might make porting simpler.

Reply:
Funny how having more tools at your disposal does not translate to writing better code. Will create a small node PR for that later.